### PR TITLE
Factory method refactor

### LINKED ISSUE #158: Factory methods and class names refactoring
After a long discussion, we decided to rename some methods and class names in the Kuzzle SDKs.

## Class Names

- [x] `KuzzleDataCollection` :point_right: `Collection`
- [x] `KuzzleDataMapping` :point_right:  `CollectionMapping`
- [x] `KuzzleRole` :point_right: `Role`
- [x] `KuzzleProfile` :point_right: `Profile`
- [x] `KuzzleUser` :point_right:  `User`
- [x] `KuzzleRoom` :point_right: `Room`
- [x] `KuzzleMemoryStorage` :point_right: `MemoryStorage`
- [x] `KuzzleSecurity` :point_right: `Security`
- [x] `KuzzleDocument` :point_right: `Document`

## Methods

- [x] `dataCollectionFactory` :point_right: `collection`
- [x] `roomFactory` :point_right: `room`
- [x] `dataMappingFactory` :point_right: `collectionMapping`
- [x] `roleFactory` :point_right: `role`
- [x] `profileFactory` :point_right: `profile`
- [x] `userFactory` :point_right: `user`
- [x] `documentFactory` :point_right: `document`
- [x] `getRole` :point_right: `fetchRole`
- [x] `getProfile` :point_right: `fetchProfile`
- [x] `getUser` :point_right: `fetchUser`

### PR DESCRIPTION
**Pull request by xbill82**
Fixes #158 and introduces a whole lotta breaking changes. :heart:

### CODE REVIEW

File: src/kuzzle.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 var
   uuid = require('uuid'),
-  KuzzleDataCollection = require('./kuzzleDataCollection'),
-  KuzzleSecurity = require('./security/kuzzleSecurity'),
-  KuzzleMemoryStorage = require('./kuzzleMemoryStorage'),
-  KuzzleUser = require('./security/kuzzleUser'),
+  Collection = require('./kuzzleCollection'),
```
Comment (dbengsch) [inline src/kuzzle.js]: I'd also rename the files

Comment (xbill82) [inline src/kuzzle.js]: I'd rather leave the `kuzzle` prefix to the files, to keep this notion of namespacing we have in the code. But I've no strong opinion on it.

Comment (dbengsch) [inline src/kuzzle.js]: I don't really see why, we already renamed contructors in them. `kuzzle` in the filename does not bring anything to the comprehension, and is not aligned with constructor name.

File: src/kuzzleCollectionMapping.js [hunk 2]
```diff
@@ -13,13 +13,13 @@
  *  (currently handled by ElasticSearch), and your searches may suffer from below-average performances, depending on
  *  the amount of data you stored in a collection and the complexity of your database.
  *
- *  The KuzzleDataMapping object allow to get the current mapping of a data collection and to modify it if needed.
+ *  The CollectionMapping object allow to get the current mapping of a data collection and to modify it if needed.
  *
- * @param {object} kuzzleDataCollection - Instance of the inherited KuzzleDataCollection object
+ * @param {object} kuzzleDataCollection - Instance of the inherited Collection object
```
Comment (ballinette) [inline src/kuzzleCollectionMapping.js]: `@param {object} Collection`

Comment (ballinette) [inline src/kuzzleCollectionMapping.js]: (and then rename all instances of `kuzzleDataCollection` to `collection` within the method)

File: src/kuzzleSubscribeResult.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**
- * Sugar-code handling the result of a KuzzleRoom.renew call
+ * Sugar-code handling the result of a Room.renew call
  * @constructor
  */
 function KuzzleSubscribeResult() {
```
Comment (scottinet) [inline src/kuzzleSubscribeResult.js]: This class (and the file containing it) needs to be renamed.

File: src/security/kuzzleProfile.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 var
   KuzzleSecurityDocument = require('./kuzzleSecurityDocument');
```
Comment (scottinet) [inline src/security/kuzzleProfile.js]: Hasn't this been renamed into `SecurityDocument`?

File: src/security/kuzzleProfile.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
 
-function KuzzleProfile(kuzzleSecurity, id, content) {
+function Profile(kuzzleSecurity, id, content) {
```
Comment (scottinet) [inline src/security/kuzzleProfile.js]: `kuzzleSecurity` must be renamed

File: src/security/kuzzleProfile.js (hunk 4 above)
Comment (scottinet) [inline src/security/kuzzleProfile.js]: This should also apply to this variable name. And this modification should be applied to all subsequent files.

File: src/security/kuzzleProfile.js (hunk 5 above)
Comment (xbill82) [inline src/security/kuzzleProfile.js]: `kuzzleSecurity` here is a variable name. I think that variable names are out of the scope of this refactoring.

**Comment by xbill82** [comment]
Ok fellows, I see that the comments are related to file and variable names. I'm afraid we're pushing things too far here. This whole thing started with a simple method renaming and it grew into something affecting the whole repo.

I'm not against doing it, just double checking if it's not a little bit too much.

**Comment by scottinet** [comment]
Here is my personal opinion:

* file names: the same way Java enforces files to have the same name than the public class exposed in it, we should name our files to reflect what's exposed in them. It's a matter of maintenability, even if we simply remove a prefix. To me, this is a mandatory step in this refactor.

* variable names are more debatable, especially in private code: as long as these variables retain their meaning (and with this refactor, they do), we might or might not rename them. I find it cleaner to have shorter variable names, the same way I find it cleaner to have shorter function names. And I think it's better if our variable names follow the same convention than our functions and classes. But I won't block this PR over this.

File: src/kuzzle.js (hunk 1 above)
Comment (xbill82) [inline src/kuzzle.js]: Ok, so I propose to follow the same naming conventions as other SDK: let's make id TitleCase.

* Kuzzle.js
* Collection.js
* CollectionMapping.js
* Document.js,
* MemoryStorage.js
* Room.js
* SearchResult.js
* SubscribeResult

**Comment by xbill82** [comment]
@dbengsch @scottinet renamed all the files :sweat:

**Pull request merged**